idf_build_apps/main.py

```diff
@@ -674,14 +675,34 @@ def get_parser() -> argparse.ArgumentParser:
         help='Path to the junitxml file. If specified, the junitxml file will be generated. Can expand placeholder @p',
     )
 
+    completions_parser = actions.add_parser(
+        'completions',
+        help='Add the autocompletion activation script to the shell rc file. '
+        'Run `idf-build-apps complations --help` for more information on a command',
+        description='Without argument print instructions for manual activation. '
+        'With argument add the autocompletion activation script to the shell rc file for bash, zsh, and fish. '
+        'Other shells are not supported',
+        formatter_class=IdfBuildAppsCliFormatter,
+    )
+    completions_parser.add_argument(
+        '-a', '--activate', action='store_true', help='Activate autocompletion automatically and permanently'
+    )
+
```

Comment:
I recommend adding one more argument here, 

```python
'--shell', choices=['auto', 'bash', 'zsh', 'fish'], default='auto'
```

so users could specify something if `$SHELL` not exists.